### DISCUSSION TITLE
How do I set `colorcolumn` based on prettier's config?

### DISCUSSION THREAD

**Discussion by rafa-hover** [LazyVim/LazyVim#2964] [General]
Is it possible to automatically identify what the max text width is by reading prettier's config and automatically mapping that to `colorcolumn`?

**Comment by dpetka2001** [comment]
You can use `vim.fn.system` to run Linux utils and extract the value you want from the file you want. You can then assign that to `vim.opt.colorcolumn`. The following should probably work
```lua
-- ~/.config/nvim/lua/config/options.lua

local path = vim.fs.find(".prettierrc", { path = vim.fn.expand("%:p:h"), upward = true, stop = vim.uv.os_homedir() })

if not vim.tbl_isempty(path) then
  local col = vim.fn.system("sed -n '/printWidth/p' " .. path[1] .. "| awk -F':|,' '{print $2}'| cut -d' ' -f2")
  if col == "" then
    return
  else
    vim.opt.colorcolumn = string.format("%d", col)
  end
end
```
Play around with the Linux utils on your terminal to verify you actually get the value you want to extract.
